### PR TITLE
feat(security-agent): complete security agent and code reviewer loop (w6a)

### PR DESCRIPTION
**Pull request by iscekic**
## Summary

1. The Security Agent now sends a push when an analysis completes or fails, and when a remediation is queued, opens a pull request, fails, is blocked, needs no changes, or is cancelled.
2. The mobile app now opens the Security Agent audit report in a native screen that shows the report period and each finding's activity.
3. The mobile app now has a Code Reviewer "Review memory" screen with paginated proposals, and owners can enable review memory there.
4. Auto-remediation now requires approval by default, so no remediation pull request opens on its own while that setting is on.
5. A finding's detail view now shows an ordered timeline of every remediation attempt.
6. The notifications screen now greys out categories that do not apply to the account and shows the reason under each one.
7. Agent messages that update files now list the updated files.
8. The "needs input" badge on agent sessions now stays dismissed after the app restarts.
9. Repository selection for Code Reviewer on mobile now saves each toggle automatically.
10. Replacing an existing organization GitLab connection now requires an owner or billing manager.

---

The security-auto-analysis worker now settles an admitted `security`-domain operation-ledger row on every terminal remediation transition and dispatches a typed `security_lifecycle` push to the finding owner. The settle joins by `provider_ref = attemptId` and is best-effort post-commit, so a settle or push failure never rolls back the terminal persist. The push producer resolves a personal owner to the owner user and an organization owner to the organization owners, then posts the web internal notifications route with a 10-second abort.

<details>
<summary>Files</summary>

- `services/security-auto-analysis/src/remediation.ts` — admits and settles the remediation ledger row; adds the lifecycle push producers and wires them into every terminal path.
- `services/security-auto-analysis/src/callbacks.ts` — runs auto-remediation as a best-effort post-completion follow-up and dispatches `analysis_completed` and `analysis_failed` pushes.
- `services/security-auto-analysis/src/types.ts` — adds the approval flag to the worker config schema and default.

</details>

The `security_lifecycle` push type is added end to end. The push data schema, the internal dispatch request, the Android channel and generic preview copy, the notifications worker dispatch core, and the web internal notifications route all carry the new variant. The route dispatches lifecycle pushes after the response, so push latency never holds the producer's 10-second abort, and mobile taps route lifecycle pushes to the finding detail.

<details>
<summary>Files</summary>

- `packages/notifications/src/push-data.ts` — adds the `security_lifecycle` variant and event enum.
- `packages/notifications/src/rpc-schemas.ts` — adds the lifecycle internal dispatch request schema and event enum.
- `packages/notifications/src/push-presentation.ts` — maps lifecycle pushes to the security channel and generic copy.
- `services/notifications/src/lib/internal-dispatch-push.ts` — dispatches lifecycle pushes to the recipient list with the preference gate.
- `apps/web/src/lib/notifications-worker-client.ts` — adds the lifecycle dispatch client.
- `apps/web/src/app/api/internal/security-agent/notifications/route.ts` — accepts the lifecycle body and dispatches push after the response.
- `apps/mobile/src/lib/notification-path.ts` — routes lifecycle pushes to the finding detail.

</details>

The manual sync, dismissal, analysis-start, and apply-auto-remediation procedures now admit `security`-domain ledger rows and settle them from terminal command state, emitting `security_command_settled`. A new bounded `getCommandStatuses` batch procedure reads up to 100 command ids and settles each terminal command; the single `getCommandStatus` stays for older clients. The findings list response now nulls the raw Dependabot alert JSON, and the analysis detail response adds an ordered remediation timeline.

<details>
<summary>Files</summary>

- `apps/web/src/lib/security-agent/router/shared-handlers.ts` — adds the ledger admission and settle helpers, the batch status handler, the list DTO, and the remediation timeline query.
- `apps/web/src/lib/security-agent/db/security-commands.ts` — adds the batch status reader.
- `packages/db/src/security-agent-command-repository.ts` — adds the batch command reader.
- `packages/db/src/index.ts` — exports the batch reader.
- `apps/web/src/lib/security-agent/core/schemas.ts` — adds the batch input schema and the approval save field.
- `apps/web/src/routers/security-agent-router.ts` — registers the batch procedure.
- `apps/web/src/routers/organizations/organization-security-agent-router.ts` — registers the organization batch procedure.
- `apps/web/src/components/security-agent/security-agent-command-copy.ts` — reuses the shared command-type union.

</details>

Code review creation now admits a `code_review`-domain operation-ledger row, and every terminal transition settles it, emitting one `code_review_settled` outbox event. The settle is a compare-and-set, so callback redelivery and the stale-review reaper are idempotent, and the reaper settles inside its terminalize transaction so a settle failure rolls back the claim. The list row type drops the heavy council result, manual config, and previous summary columns.

<details>
<summary>Files</summary>

- `apps/web/src/lib/code-reviews/code-review-ledger.ts` — new; maps terminal status to ledger outcome and settles with the outbox event.
- `apps/web/src/lib/code-reviews/db/code-reviews.ts` — admits the ledger row at both creation paths and narrows the list row type.
- `apps/web/src/app/api/internal/code-review-status/[reviewId]/route.ts` — settles on completion, terminal short-circuits, and non-analytics terminal updates.
- `apps/web/src/lib/code-reviews/reap-stale-reviews.ts` — settles inside the terminalize transaction.
- `apps/web/src/lib/code-reviews/core/schemas.ts` — removes the old list response types.
- `packages/db/src/operation-ledger.ts` — adds the `code_review` domain.
- `apps/web/src/app/api/webhooks/bitbucket/[integrationId]/route.ts` — admits and settles ledger rows for webhook-created and superseded reviews.
- `apps/web/src/lib/integrations/platforms/bitbucket/manual-code-review-trigger.ts` — admits and settles ledger rows for manually triggered reviews.

</details>

The shared analytics event map gains the `code_review_settled` and `security_command_settled` terminal events with the DEC-05 base fields, plus the `security` and `code_review` intent enums. The durable outbox validates every settle payload against these strict schemas.

<details>
<summary>Files</summary>

- `packages/app-shared/src/analytics/event-map.ts` — adds the two events, the intent enums, and the command-type-to-intent map.

</details>

Auto-remediation now defaults to requiring approval. The policy rejects automatic admission with `approval_required` when the flag is set, and a manual start is the approval path. The include-existing bulk command is skipped while approval is required, because every candidate would be rejected. The web and mobile settings forms expose the toggle, and the config schema, save input, and capability copy all carry the new flag.

<details>
<summary>Files</summary>

- `packages/worker-utils/src/security-remediation-policy.ts` — adds the flag to the config type and the `approval_required` rejection reason.
- `apps/web/src/lib/security-agent/core/types.ts` — adds the schema field with a true default.
- `apps/web/src/lib/security-agent/core/constants.ts` — adds the default value.
- `apps/web/src/lib/security-agent/db/security-config.ts` — skips the include-existing remediation command when approval is required.
- `apps/web/src/lib/security-agent/db/security-remediation.ts` — passes the flag into the policy config.
- `apps/web/src/components/security-agent/security-config-types.ts` — adds the form state, source, and save payload fields.
- `apps/web/src/components/security-agent/SecurityConfigForm.tsx` — renders the approval switch and guards unsaved changes.
- `apps/web/src/components/security-agent/SecurityConfigPage.tsx` — hydrates the field from the config.
- `apps/web/src/components/security-agent/SecurityAgentContext.tsx` — threads the field through config save.
- `packages/app-shared/src/security-agent/presentation.ts` — adds the `approval_required` unavailable copy.
- `apps/mobile/src/components/security-agent/automation-settings-screen.tsx` — renders the approval toggle and the enable confirmation.

</details>

`organizations.withMembers` now returns a role-gated union: member callers get a stripped shape without the Stripe customer id, the invitation secret, or per-member daily usage, while admin-and-above keep the full shape. The code review list and detail similarly null internal ledger and check-run identifiers for non-admin organization callers. Every consumer migrates to the union type and guards the stripped fields.

<details>
<summary>Files</summary>

- `apps/web/src/lib/organizations/organization-types.ts` — adds the member and admin response variants and the public member schemas.
- `apps/web/src/lib/organizations/organizations.ts` — no behavior change; type-only import updates.
- `apps/web/src/routers/organizations/organization-router.ts` — returns the role-gated member shape.
- `apps/web/src/routers/code-reviews/code-reviews-router.ts` — nulls internal identifiers for non-admin callers.
- `apps/web/src/app/(app)/cloud/mcp-gateway/McpGatewayDetailContent.tsx` — migrates to the union type.
- `apps/web/src/components/organizations/FreeTrialWarningBanner.tsx` — migrates to the union type.
- `apps/web/src/components/organizations/FreeTrialWarningDialog.tsx` — migrates to the union type.
- `apps/web/src/components/organizations/OrganizationContextWrapper.tsx` — migrates to the union member type.
- `apps/web/src/components/organizations/OrganizationInfoCard.tsx` — reads the Stripe id only on the admin variant.
- `apps/web/src/components/organizations/OrganizationMembersCard.tsx` — guards the stripped invite URL and migrates to the union type.
- `apps/web/src/components/organizations/SSOSignupCard.tsx` — migrates to the union type.
- `apps/web/src/components/organizations/groups/drawer/GroupDetailsPanel.tsx` — narrows active members through the union.
- `apps/web/src/components/organizations/members/EditDailyUsageLimitUsdDialog.tsx` — migrates to the union member type.
- `apps/web/src/components/organizations/members/MemberRoleDropdown.tsx` — migrates to the union member type.
- `apps/mobile/src/components/organization/invited-member-row-state.ts` — gates the share action on the invite URL and adds resend.
- `apps/mobile/src/components/organization/invited-member-row.tsx` — offers share only when the invite URL is present.
- `apps/mobile/src/components/organization/member-limit-sheet.tsx` — silences the update toast for inline errors.
- `apps/mobile/src/components/organization/members-screen.tsx` — no behavior change; type-only updates.
- `apps/mobile/src/lib/hooks/use-organization-mutations.ts` — preserves the union variant in optimistic updates.
- `apps/mobile/src/lib/hooks/use-organization-queries.ts` — derives the union member types.
- `apps/mobile/src/components/code-reviewer/review-detail-screen.tsx` — reads the redacted-ids flag from the detail response.
- `apps/mobile/src/components/code-reviewer/review-detail-sections.tsx` — renders the gate check run as hidden when redacted.

</details>

The mobile command observer now polls tracked Security Agent commands through one bounded `getCommandStatuses` batch for the first 100 ids, with per-command overflow and a per-command fallback for older servers. A `security_lifecycle` push, app foreground, or reconnect invalidates the affected findings and command-status queries for the scope. The shared command helpers define the batch split, the missing-procedure signature, and the per-command failure copy.

<details>
<summary>Files</summary>

- `apps/mobile/src/lib/hooks/use-security-agent-commands.ts` — batch polling, overflow fallback, and scope-key invalidation.
- `apps/mobile/src/lib/security-agent.ts` — batch limit, split, missing-procedure detection, and reconciliation.
- `apps/mobile/src/lib/hooks/use-security-findings.ts` — tracks the command id only when present.
- `apps/mobile/src/lib/hooks/use-security-lifecycle-invalidation.ts` — subscribes to push, foreground, and reconnect invalidation.
- `apps/mobile/src/app/(app)/_layout.tsx` — mounts the lifecycle invalidation hook.
- `packages/app-shared/src/security-agent/commands.ts` — shared command types, invalidation scopes, and failure copy.

</details>

The mobile app now renders the Security Agent audit report natively, with loading, empty, retryable, and non-retryable states. The organization procedure is billing-gated, so a forbidden viewer sees a non-retryable empty state.

<details>
<summary>Files</summary>

- `apps/mobile/src/components/security-agent/audit-report-screen.tsx` — renders the report summary and per-finding collapsible sections.
- `apps/mobile/src/app/(app)/(tabs)/(3_profile)/security-agent/[scope]/audit-report.tsx` — the route.
- `apps/mobile/src/components/security-agent/audit-report-button.tsx` — header action that opens the native report.

</details>

Review memory gains a paginated `listProposalsPage` procedure with keyset cursors, and the mobile app renders a Review memory screen with loading, empty, off, and error states plus an enable action for owners. The array-shaped `listProposals` stays for the web panel and stale clients.

<details>
<summary>Files</summary>

- `apps/web/src/lib/code-reviews/review-memory/db.ts` — adds the paginated list with cursor encode and decode.
- `apps/web/src/routers/code-reviews/review-memory-router.ts` — adds the `listProposalsPage` procedure.
- `packages/trpc/src/mobile.ts` — exposes the review memory router to mobile.
- `apps/mobile/src/components/code-reviewer/review-memory-screen.tsx` — renders the paginated proposals and enable action.
- `apps/mobile/src/app/(app)/(tabs)/(3_profile)/code-reviewer/[scope]/review-memory.tsx` — the route.
- `apps/mobile/src/components/code-reviewer/platform-overview-rows.ts` — adds the review memory overview row.
- `apps/mobile/src/components/code-reviewer/platform-overview-screen.tsx` — wires the review memory navigation.

</details>

Repository selection for Code Reviewer now saves as a debounced delta patch instead of a full-array overwrite, so rapid toggles are not lost and a refetch cannot clobber pending intent. The personal and organization patch procedures accept `selectedRepositoryDelta` and reject sending both the delta and the full array. The mobile overview keeps the shared numeric-id narrowing for the personal path.

<details>
<summary>Files</summary>

- `apps/mobile/src/lib/hooks/use-code-reviewer-repo-selection.ts` — debounced delta send with rollback and refetch resync.
- `apps/mobile/src/lib/hooks/use-code-reviewer.ts` — numeric-id narrowing, review memory hooks, and the delta save path.
- `apps/mobile/src/app/(app)/(tabs)/(3_profile)/code-reviewer/[scope]/[platform]/(edit)/repos.tsx` — uses the debounced toggle.
- `apps/web/src/routers/code-reviews-router.ts` — adds the personal `selectedRepositoryDelta` patch support.
- `apps/web/src/routers/organizations/organization-code-reviews-router.ts` — adds the organization `selectedRepositoryDelta` patch support.

</details>

The notification preferences response now includes a per-category capability map, and the mobile notifications screen disables categories that do not apply with the server's reason. Balance alerts require an organization, security findings require an enabled Security Agent, and KiloClaw activity requires an instance.

<details>
<summary>Files</summary>

- `apps/web/src/routers/user-router.ts` — computes the capability map and returns it.
- `apps/mobile/src/components/notifications-screen.tsx` — renders the per-category disabled state and reason.

</details>

Session attention acknowledgements are now durable: acks persist to the mobile encrypted KV store under one storage key and survive app restarts, with a 7-day expiry and same-session re-raise handling. The store hydrates lazily so the synchronous API stays free of the native SQLCipher chain.

<details>
<summary>Files</summary>

- `apps/mobile/src/lib/session-attention.ts` — durable ack store with lazy hydration and reconciliation.
- `apps/mobile/src/lib/storage-keys.ts` — adds the storage key.

</details>

Standalone patch parts now render in the mobile transcript as a file-count summary with the file list, instead of rendering nothing. The visibility predicate counts a patch part as content only when it carries files.

<details>
<summary>Files</summary>

- `apps/mobile/src/components/agents/part-renderer.tsx` — renders the patch summary and file list.
- `apps/mobile/src/components/agents/part-types.ts` — adds the patch part guard.
- `apps/mobile/src/components/agents/message-visibility.ts` — counts patch parts with files as content.

</details>

Replacing an existing organization GitLab integration now requires a billing role, while a first-time connect keeps member-level access. Self-hosted GitLab OAuth requires custom client credentials, and the connect and callback handlers surface the new denial codes.

<details>
<summary>Files</summary>

- `apps/web/src/lib/integrations/oauth/common.ts` — adds the access-denial code mapping.
- `apps/web/src/lib/integrations/oauth/platforms/gitlab-connect.ts` — billing-gates replacement and stores custom credentials.
- `apps/web/src/lib/integrations/oauth/platforms/gitlab-callback.ts` — re-checks the billing gate and loads cached credentials.
- `apps/web/src/components/integrations/GitLabIntegrationDetails.tsx` — maps the new error codes to copy.

</details>

The mobile finding detail renders the ordered remediation timeline returned by the analysis procedure, with local labels for each remediation audit action and a fallback to the raw action string.

<details>
<summary>Files</summary>

- `apps/mobile/src/components/security-agent/finding-remediation-panel.tsx` — renders the timeline and attempt history.

</details>

Tests: 52 test files changed across web, mobile, packages, and services.
Generated: none.

---

## Verification

Two verification rounds ran three cases on iOS.

| Case | What it proves | Platform | Result |
|---|---|---|---|
| S1 | The Security Agent audit report screen renders with an empty state, and the back control returns. | ios | passed |
| S2 | The Review memory screen renders from the GitHub overview, and the back control returns. | ios | passed after the fix in round 2 |
| S3 | The Notifications screen shows disabled categories with their reason lines. | ios | passed |

Defects reproduced on the unfixed build:

- verify-r1: the Review memory row was absent from the Code Reviewer overview. The seeded account had no GitHub installation, and the row is GitHub-only. The fix seeded installation 144771093. The defect does not still reproduce in verify-r2.

### Security Agent audit report screen

The user now sees an Audit report screen with a "No recorded activity" empty state and a date-range reason line. The picture shows the "Audit report" header at the top left and the empty-state text with the date range in the middle of the screen.

### Review memory screen

The user now sees a Review memory screen with an off state, a reason line, and an "Enable review memory" button. The picture shows the "Review memory" header under "CODE REVIEWER" at the top left and the off state with the button below it.

### Notifications screen

The user now sees disabled notification categories, each with a reason line. The picture shows the dimmed "Balance alerts" and "Security findings" rows in the CATEGORIES list, each with its reason line under the label.

## Reviewer Notes

Notes: none.

Human steps: none.

### CODE REVIEW

File: apps/web/src/routers/code-reviews/code-reviews-router.ts [hunk 1]
```diff
@@ -248,8 +247,30 @@ export const codeReviewRouter = createTRPCRouter({
...
+        // - cli_session_id: internal CLI session id used to look up the billing
+        //   ledger (getSessionUsageFromBilling) -> null for non-admin.
+        // - dispatch_reservation_id: internal dispatch reservation id -> null
+        //   for non-admin.
+        // - check_run_id: internal GitHub Check Run ID -> null for non-admin.
+        // total_cost_musd (display cost) is retained: the mobile review detail
+        // screen renders it (review-detail-screen.tsx).
+        const callerRole = await ensureOrganizationAccess(ctx, fullInput.organizationId);
+        const canSeeRawIds = callerRole === 'owner' || callerRole === 'admin';
+        const visibleReviews = canSeeRawIds
```
Comment (pandemicsyn) [inline apps/web/src/routers/code-reviews/code-reviews-router.ts:263]: bot: Organization-member redaction is bypassable through the Code Review detail endpoint.

Suggested fix: Use the role returned by `ensureOrganizationAccess` in `get` as well, and null `session_id`, `cli_session_id`, `dispatch_reservation_id`, and `check_run_id` (and any equivalent identifiers returned through attempts) before returning an organization review to a non-owner/non-admin. Add a regression test that calls `get` as an ordinary member and asserts those fields are absent/null.

File: apps/web/src/routers/organizations/organization-code-reviews-router.ts [hunk 2]
```diff
@@ -852,6 +869,18 @@ export const organizationReviewAgentRouter = createTRPCRouter({
             | Array<number | string>
             | undefined) || [];
 
+        // Delta repository selection: `selectedRepositoryDelta` computes
+        // next = (stored ∪ add) \ remove, with remove winning on overlap.
+        // The full-array `selectedRepositoryIds` input is kept for
+        // compatibility and flows through the merge helper unchanged.
+        let selectedRepositoryIdsFromDelta: Array<number | string> | undefined;
+        if (input.selectedRepositoryDelta !== undefined) {
```
Comment (pandemicsyn) [inline apps/web/src/routers/organizations/organization-code-reviews-router.ts:875]: bot: Bitbucket repository deltas persist unvalidated IDs and can leave selected mode with no repository.

Suggested fix: After calculating the effective delta result for Bitbucket, load readiness and pass `{ repositorySelectionMode, selectedRepositoryIds: effectiveSelectedRepositoryIds }` through `requireBitbucketRepositorySelection` before `upsertAgentConfig`. Add tests for an unknown UUID, an empty result after removal, duplicate IDs, and unavailable repository cache.

**Review by pandemicsyn — CHANGES_REQUESTED** [review]

File: apps/web/src/routers/code-reviews/code-reviews-router.ts (hunk 1 above)
Comment (iscekic) [reply apps/web/src/routers/code-reviews/code-reviews-router.ts:263]: (bot) Fixed in f670eaa84. The get procedure now redacts session_id, cli_session_id, dispatch_reservation_id, and check_run_id on the review, and session_id/cli_session_id/execution_id on attempts, for non-owner/non-admin org members. Regression test added in code-reviews-router.test.ts.

File: apps/web/src/routers/organizations/organization-code-reviews-router.ts (hunk 2 above)
Comment (iscekic) [reply apps/web/src/routers/organizations/organization-code-reviews-router.ts:875]: (bot) Fixed in dfa87982b. patchReviewConfig now validates Bitbucket effective selections through requireBitbucketRepositorySelection before upsertAgentConfig. Tests cover unknown UUID, empty-after-removal, duplicate IDs, and unavailable cache.

**Review by eshurakov — APPROVED** [review]

File: apps/web/src/lib/security-agent/router/shared-handlers.ts [hunk 3]
```diff
@@ -1725,6 +1924,17 @@ export function createSecurityAgentHandlers<TExtra = {}>(deps: SecurityAgentDeps
         });
         if (!queued.queued) return { success: false, ...queued };
 
+        // Admit the remediation ledger row after the attempt committed so the
+        // Worker's terminal callback can settle it. Best-effort: the attempt
+        // already exists, so a ledger write failure must not fail the request.
+        await admitSecurityRemediationLedgerRow({
```
Comment (pandemicsyn) [inline apps/web/src/lib/security-agent/router/shared-handlers.ts]: bot: Manual remediation can finish before its ledger row exists, leaving the row permanently unsettled.

Suggested fix: Admit the operation-ledger row before submitting/queueing the remediation, or make admission reconcile the attempt's current terminal state and settle immediately when it has already completed. Keep failures best-effort if desired, but ensure a delayed successful admission cannot create an admitted row after the only terminal callback has already skipped settlement. Add a regression test that forces terminal callback processing before the admission completes.

File: apps/web/src/routers/organizations/organization-code-reviews-router.ts [hunk 4]
```diff
@@ -852,6 +867,18 @@ export const organizationReviewAgentRouter = createTRPCRouter({
...
```
Comment (pandemicsyn) [inline apps/web/src/routers/organizations/organization-code-reviews-router.ts:875]: bot: Repository-selection deltas bypass GitHub/GitLab numeric-ID validation and can persist unusable selections.

Suggested fix: Run `requireRepositoryIdsForPlatform(platform, effectiveSelectedRepositoryIds)` before `upsertAgentConfig` whenever a full selection or delta is supplied, and persist the validated output. Add tests rejecting string IDs in GitHub and GitLab full-array and delta PATCH requests; retain the existing Bitbucket readiness validation.

File: apps/mobile/src/lib/session-attention.ts [hunk 5]
```diff
@@ -1,42 +1,221 @@
...
+  return JSON.stringify(entries);
+}
+
+async function writeEntriesSafely(serialized: string): Promise<void> {
+  const kv = await loadKv();
+  if (!kv) {
+    return;
+  }
+  try {
+    await kv.setItem(SESSION_ATTENTION_KEY, SESSION_ATTENTION_ENTRY_KEY, serialized);
```
Comment (pandemicsyn) [inline apps/mobile/src/lib/session-attention.ts:129]: bot: Persisted session-attention acknowledgements are shared across accounts and survive sign-out.

Suggested fix: Include the authenticated user ID in the persistence scope/key and reset the in-memory attention store when that identity changes, or explicitly delete `SESSION_ATTENTION_KEY` and clear the store during sign-out. Add an account-switch test that acknowledges a session as one user, signs out, hydrates as another user, and verifies the prior acknowledgement cannot suppress a reused session ID.

**Review by pandemicsyn — APPROVED** [review]
lgtm, just a couple minor ones from the robot. The manual remediation one seems very edge case but might be worth a peek.

File: apps/web/src/lib/security-agent/router/shared-handlers.ts (hunk 3 above)
Comment (iscekic) [reply apps/web/src/lib/security-agent/router/shared-handlers.ts]: Valid. Fixed in f0ff68f3a.

The web handler admitted the row after the Worker fetch returned. The queue consumer can reach a terminal state in milliseconds (`blockAttempt`, launch failure), and the terminal settle joins on `provider_ref = attemptId`, so a late admit left the row admitted forever.

Root-cause fix instead of a reconcile: the Worker now admits the row in `startManualRemediation` before `enqueueRemediationAttempt`, so no terminal callback can precede the admit. This matches the auto path (`maybeAdmitAutoRemediationForCompletedAnalysis`) and deletes the web-side duplicate (`admitSecurityRemediationLedgerRow`, `resolveSecurityLedgerUserId`, `securityRemediationLedgerResourceKey`, both call sites). A queue-send failure now settles the admitted row through `settleAdmittedLedgerRowBestEffort`.

Tests: `remediation.test.ts` asserts the order `['admit', 'accept', 'enqueue']` and that an admit failure never fails the start; `security-ledger-settled-outcomes.test.ts` asserts the web handler admits nothing.

File: apps/web/src/routers/organizations/organization-code-reviews-router.ts (hunk 4 above)
Comment (iscekic) [reply apps/web/src/routers/organizations/organization-code-reviews-router.ts:875]: Valid. Fixed in f0ff68f3a.

`saveReviewConfig` called `requireRepositoryIdsForPlatform`; `patchReviewConfig` did not, and the patch schema accepts `z.union([z.number(), z.string()])`. A GitHub or GitLab PATCH could persist string ids the trigger can never match.

`patchReviewConfig` now runs `requireRepositoryIdsForPlatform(platform, ...)` on the effective selection whenever the patch supplies a full array or a delta, and persists the validated output. A patch that touches no selection skips the check, so an unrelated edit cannot fail on an already-stored value. The Bitbucket readiness validation is unchanged.

Tests: four new cases reject string ids for GitHub and GitLab, in both the full-array and the delta form, and assert the stored selection is untouched.

File: apps/mobile/src/lib/session-attention.ts (hunk 5 above)
Comment (iscekic) [reply apps/mobile/src/lib/session-attention.ts:129]: Valid, fixed in f0ff68f3a — with one correction to the severity.

The scope `session-attention` carried no user id and had no sign-out cleanup, so one account's session ids stayed on the device and were hydrated by the next account. Every peer durable store is either per-user scoped (`cache:<userId>:`) or deleted at sign-out, so this one was the outlier.

Correction: cross-account suppression is not practically reachable. Session ids are UUIDs, so account B never holds one of account A's ids. The real defect is that account A's session ids survive the teardown.

Fix: `clearSessionAttentionForSignOut` clears the in-memory store, chains the KV `clearScope` behind the write FIFO so a queued bump cannot re-persist the blob, and re-arms hydration. It joins the `Promise.allSettled` sign-out batch in `auth-context.tsx` and never throws.

Tests: an account-switch test acks a session, signs out, hydrates again, and asserts the badge returns; plus the queued-bump race and the KV-failure path.
